services/tenant-management-service/src/providers/idp/idp-auth0.provider.ts

```diff
@@ -0,0 +1,170 @@
+import {Provider} from '@loopback/context';
+
+import {ConfigureIdpFunc, IdpDetails} from '../../types';
+import {ManagementClient} from 'auth0';
+
+import {Auth0Response, ConfigValue, OrganizationData, UserData} from './types';
+
+import {TenantConfigRepository} from '../../repositories/tenant-config.repository';
+import {repository} from '@loopback/repository';
+
+import {HttpErrors} from '@loopback/rest';
+
+export class Auth0IdpProvider
+  implements Provider<ConfigureIdpFunc<Auth0Response>>
+{
+  management: ManagementClient;
+
+  constructor(
+    @repository(TenantConfigRepository)
+    private readonly tenantConfigRepository: TenantConfigRepository,
+  ) {}
+
+  value(): ConfigureIdpFunc<Auth0Response> {
+    return payload => this.configure(payload);
+  }
+  async configure(payload: IdpDetails): Promise<Auth0Response> {
+    this.management = new ManagementClient({
+      domain: process.env.AUTH0_DOMAIN ?? '',
+      clientId: process.env.AUTH0_CLIENT_ID ?? '',
+      clientSecret: process.env.AUTH0_CLIENT_SECRET ?? '',
+      audience: process.env.AUTH0_AUDIENCE,
+    });
+    const {tenant} = payload;
+    const planTier = tenant.plan.tier;
+    const tenantConfig = await this.tenantConfigRepository.findOne({
+      where: {tenantId: tenant.id},
+    });
+    if (!tenantConfig) {
+      throw new HttpErrors.NotFound(
+        `Tenant configuration not found for tenant: ${tenant.id}`,
+      );
+    }
+    const configValue: ConfigValue = tenantConfig.configValue;
+    if (!configValue) {
+      throw new HttpErrors.NotFound(
+        `Tenant configuration not found for tenant: ${tenant.id}`,
+      );
+    }
+
+    const organizationData: OrganizationData = {
```

Comment:
tenant_config depends on two things 
tenant_id and config_key 
please take key also into account as I can store diff types of config for a tenant